docs/architecture.md

```diff
@@ -0,0 +1,23 @@
+# Architecture
+
```

Comment:
TODO add doc about how NAC integrates with OADP

Reply:
add also requirements to command work (ex.: need to already have `RELATED_IMAGE_NON_ADMIN_CONTROLLER` in manager.yaml)